src/internal/types.ts

```diff
@@ -96,15 +97,30 @@ type IsBoundedString<T> = T extends string
  *
  * @see EnumerableStringKeyedValueOf
  */
-export type EnumerableStringKeyOf<T> = `${Exclude<keyof T, symbol>}`;
+export type EnumerableStringKeyOf<T> =
+  Required<T> extends Record<infer K, unknown>
+    ? `${Exclude<K, symbol>}`
+    : never;
+
+/**
+ * Extracts the value type from an object type T.
+ */
+export type ValuesOf<T> =
```

Comment:
This isn't used outside this file so it shouldn't be exported. We also usually put non-exported "private" types below the exported ones so it's easier to read through the file.